From 3f0e2bcb5776274c3403ca41ee723a59d71e6da1 Mon Sep 17 00:00:00 2001
Message-Id: <3f0e2bcb5776274c3403ca41ee723a59d71e6da1.1423872525.git.feitong.yi@intel.com>
In-Reply-To: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
References: <3062c7e6b919c393846ff0e13558e7f648b352ca.1423872525.git.feitong.yi@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sat, 31 Jan 2015 08:52:49 -0500
Subject: [PATCH 82/95] MUST_REBASE [VPG]: video/adf/intel: Fix duplicate
 modesets on dsi

During dsi modeset call we are not checking if modeset already done.
This patch addes the check the modeset by verifying DPMS State.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we
need to work on i915 and ADF convergence path before it can be
upstreamed

Issue: GMINL-5501
Change-Id: I6e3fc03f02730462bfbf6ad2bfd6cc7226209e4f
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 .../video/adf/intel/core/common/dsi/gen_dsi_pipe.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
index 6e29112..753d67c 100644
--- a/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
+++ b/drivers/video/adf/intel/core/common/dsi/gen_dsi_pipe.c
@@ -376,6 +376,12 @@ static int dsi_modeset(struct intel_pipe *pipe,
 		return err;
 	}
 
+	/* Avoid duplicate modesets */
+	if (dsi_pipe->dpms_state == DRM_MODE_DPMS_ON) {
+		pr_info("ADF: %s: DSI already enabled\n", __func__);
+		return err;
+	}
+
 	mutex_lock(&config->ctx_lock);
 	dsi_display_off(pipe);
 	dsi_display_on(pipe, mode);
-- 
1.7.9.5

